flash/core/data/input_transform.py

```diff
@@ -53,231 +59,129 @@ def wrapper(self, *args, **kwargs) -> Any:
     return wrapper
 
 
-class InputTransform(Properties):
-    def configure_transforms(self, *args, **kwargs) -> Dict[InputTransformPlacement, Callable]:
-        """The default transforms to use.
-
-        Will be overridden by transforms passed to the ``__init__``.
-        """
-
-    def configure_per_sample_transform(self, *args, **kwargs) -> Callable:
-        """The default transforms to use.
-
-        Will be overridden by transforms passed to the ``__init__``.
-        """
-        return self._identity
+# Credit to Torchvision Team:
+# https://pytorch.org/vision/stable/_modules/torchvision/transforms/transforms.html#Compose
+class Compose:
+    """Composes several transforms together.
 
-    def configure_per_batch_transform(self, *args, **kwargs) -> Callable:
-        """The default transforms to use.
+    This transform does not support torchscript.
+    """
 
-        Will be overridden by transforms passed to the ``__init__``.
-        """
-        return self._identity
-
-    def configure_collate(self, *args, **kwargs) -> Callable:
-        """The default transforms to use.
+    def __init__(self, transforms):
+        self.transforms = transforms
 
-        Will be overridden by transforms passed to the ``__init__``.
-        """
-        return default_collate
+    def __call__(self, x):
+        for t in self.transforms:
+            x = t(x)
+        return x
 
-    def configure_per_sample_transform_on_device(self, *args, **kwargs) -> Callable:
-        """The default transforms to use.
+    def __repr__(self):
+        format_string = self.__class__.__name__ + "("
+        for t in self.transforms:
+            format_string += "\n"
+            format_string += f"{t}"
+        format_string += "\n)"
+        return format_string
 
-        Will be overridden by transforms passed to the ``__init__``.
-        """
-        return self._identity
 
-    def configure_per_batch_transform_on_device(self, *args, **kwargs) -> Callable:
-        """The default transforms to use.
-
-        Will be overridden by transforms passed to the ``__init__``.
-        """
-        return self._identity
-
-    def __init__(
-        self,
-        running_stage: RunningStage,
-        transform: Union[Callable, List, Dict[str, Callable]] = None,
-        **transform_kwargs,
-    ):
+class InputTransform(Properties):
```

Comment:
If I understand correctly, this only holds the transforms. The InputTransform is not a callable (I didn't see `__call__` implemented). Have you considered a different name to differentiate between transforms that can be applied vs. the container class of holding a collection of transforms?

Reply:
Yes, but we wanted to keep Input, Output, InputTransform, OutputTransform.